src/components/AddMaterialForm/ui/Button/Button.Style.ts

```diff
@@ -0,0 +1,39 @@
+import styled, { css } from "styled-components";
+import { IButton } from "./Button";
+
+const buttonDefaultStyle = css`
+  font-size: clamp(1rem, 0.8977rem + 0.4545vw, 1.125rem);
+  color: #fff;
+  font-weight: 600;
+  text-transform: uppercase;
+
+  width: 160px;
+
+  padding: 14px;
+
+  background-color: #040eff;
+
+  border: none;
+  border-radius: 10px;
+
+  box-shadow: 0 2px 7px rgba(0, 0, 0, 0.1);
+
+  cursor: pointer;
+`;
+
+const cancelButton = css`
+  ${buttonDefaultStyle}
+  border: 1px solid #040eff;
+  color: #040eff;
+  background-color: #fff;
+  transition: all 0.2s ease-in-out;
+  :hover {
+    color: #fff;
+    background-color: #040eff;
+  }
+`;
+
+export const StyledButon = styled.button<IButton>`
+  ${buttonDefaultStyle};
+  ${(props: { isCancelVariant: any }) => props.isCancelVariant && cancelButton}
```

Comment:
A correção deste código é um pouco mais longa, você pode criar uma `interface` ou um `type` e atribuir a ele os atributos que a `props`  irá receber.

```
interface IProps {
  isCancelVariant: boolean;
}
```

em seguida, você pode atribuir esta nova interface a sua `props`:

```suggestion
  ${(props: IProps) => props.isCancelVariant && cancelButton}
```

---
**(Modificação opcional)**
Além dessas alterações, você também pode usar o `destructuring`, não é obrigatório:

```suggestion
  ${({isCancelVariant}: IProps) => isCancelVariant && cancelButton}
```

Reply:
No Button.tsx eu já tem a interface 
![image](https://user-images.githubusercontent.com/2619027/197285592-23d545c1-800b-4f91-9e22-1e8576e64a62.png)
e está importada no arquivo de estilo para poder usar a prop da variante
![image](https://user-images.githubusercontent.com/2619027/197286951-f6b6d491-5a2f-49e6-8501-35f5ac83ab69.png)

Aí só fiz a alteração destruturando como vc sugeriu
![image](https://user-images.githubusercontent.com/2619027/197286515-4a6b36b6-b59c-45d9-b757-9f43380fb160.png)